### PR TITLE
Update dataset repo to version 1.0

### PR DESCRIPTION
**Pull request by jteijema**
This PR will update the template to version 1.0, and test it's functionality.

### CODE REVIEW

**Comment by jteijema** [comment]
Will take care of  #6 #7 and #1.

**Comment by jteijema** [comment]
image

Installing this template adds the datasets to the DatasetManager.

**Comment by JTeijema** [comment]
image

The datasets show up in the front end. The following was noted:
- Filepath only takes strings because of JSON restrictions, so parsing to a string is needed for a local file path.
- The local file path option doesn't work yet and links to the wrong place. We need to figure out how to get the right path.
-  The datasets are only available in oracle mode. Shouldn't they be available in simulation or exploration?
@J535D165 

Aside from these points, the template works.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,21 @@
 # Template for extending ASReview with a new dataset
 
+<a href="https://github.com/asreview/asreview"><img
+src="https://img.shields.io/badge/ASReview-v1.0-%23ffcb05" alt="ASReview v1.0"
+/></a>
```
Comment (J535D165) [inline README.md]: Please use Markdown.

File: asreviewcontrib/dataset_name/your_dataset.py [hunk 2]
```diff
@@ -1,29 +1,60 @@
 from asreview.datasets import BaseDataSet
 from asreview.datasets import BaseDataGroup
+from pathlib import Path
 
-class YourDataGroup(BaseDataGroup):
-    group_id = "your_data_group"
-    description = "A new data group with my awesome datasets."
+
+class example_dataset_local(BaseDataSet):
```
Comment (J535D165) [inline asreviewcontrib/dataset_name/your_dataset.py]: Please see naming conventions in pep8. `ExampleDatasetLocal`

File: asreviewcontrib/dataset_name/your_dataset.py [hunk 3]
```diff
@@ -1,29 +1,60 @@
...
+    """
+    This is an example dataset that is stored locally.
+    """
+
+    import os
```
Comment (J535D165) [inline asreviewcontrib/dataset_name/your_dataset.py]: Imports should be added on top. Much better performance.

File: asreviewcontrib/dataset_name/your_dataset.py [hunk 4]
```diff
@@ -1,29 +1,60 @@
...
+from pathlib import Path
 
-class YourDataGroup(BaseDataGroup):
-    group_id = "your_data_group"
-    description = "A new data group with my awesome datasets."
+
+class example_dataset_local(BaseDataSet):
+    """
+    This is an example dataset that is stored locally.
+    """
```
Comment (J535D165) [inline asreviewcontrib/dataset_name/your_dataset.py]: See pep257. 
```suggestion
    """This is an example dataset that is stored locally.
    """
```

File: asreviewcontrib/dataset_name/your_dataset.py [hunk 5]
```diff
@@ -1,29 +1,60 @@
 from asreview.datasets import BaseDataSet
 from asreview.datasets import BaseDataGroup
+from pathlib import Path
```
Comment (J535D165) [inline asreviewcontrib/dataset_name/your_dataset.py]: Please group imports according to pep8

File: asreviewcontrib/dataset_name/your_dataset.py [hunk 6]
```diff
@@ -1,29 +1,60 @@
...
+    This is an example dataset that is stored locally.
+    """
+
+    import os
+    __filepath__ = os.getcwd()
 
     def __init__(self):
+        super().__init__(
+            dataset_id="example_dataset_local",
+            filepath=str(Path(self.__filepath__, 'data', 'your_dataset.csv')),
```
Comment (J535D165) [inline asreviewcontrib/dataset_name/your_dataset.py]: Can't you use `Path(__file__).parent`?

File: setup.py [hunk 7]
```diff
@@ -19,12 +19,12 @@
...
     python_requires='~=3.6',
     install_requires=[
-        'asreview>=0.16',
+        'asreview>=1.0',
     ],
 
     entry_points={
         "asreview.datasets": [
-            "newDataset = asreviewcontrib.dataset_name.your_dataset:YourDataGroup"
+            "example_dataset_group = asreviewcontrib.dataset_name.your_dataset:example_dataset_group", # noqa
```
Comment (J535D165) [inline setup.py]: also add remote?

**Review by J535D165 — CHANGES_REQUESTED** [review]
Thanks, few suggestions, and some styling issues.

**Comment by jteijema** [comment]
Updated the dataset file to pep8 style @J535D165

File: setup.py (hunk 7 above)
Comment (jteijema) [reply setup.py]: The example group has both examples. I think only the group needs to be assigned as entrypoint.

**Comment by JTeijema** [comment]
Note: the following code will not add a datasetgroup:
```python
class ExampleDatasetGroupTwo(BaseDataGroup):
    """This is an example dataset group."""

    group_id = "example_dataset_group"
    description = "Example dataset group"

    def __init__(self):
        """Initialize the dataset group."""

        super().__init__()
        self.append(ExampleDatasetLocal())
        self.append(ExampleDatasetRemote())
```

File: asreviewcontrib/dataset_name/your_dataset.py (hunk 6 above)
Comment (jteijema) [reply asreviewcontrib/dataset_name/your_dataset.py]: We couldn't since the setup only installed packages and skipped datafiles. I've adjusted this now.

**Comment by JTeijema** [comment]
An issue with ASReview I am running into is the lack of errors when the dataset is not filled in correctly. I just spend around 2 hours figuring out why the groups showed up empty, only to find a typo in the filepath in one of the datasets.

**Comment by jteijema** [comment]
@J535D165 ASReview still has a problem handling local files I think: 
image

```console
ERROR:root:<urlopen error unknown url type: c>
```


Moreover, the code for the filepath is currently; 
```python
            filepath=str(Path(Path(__file__).parent,
                              'data',
                              'your_dataset.csv')),
```

I think being able to give a path instead of a str would be optimal.

Giving a path results in:
```console
ERROR:asreview.webapp.start_flask:Exception on /api/datasets [GET]
Traceback (most recent call last):
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask_cors\extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\asreview\webapp\api.py", line 291, in api_demo_data_project
    response = jsonify(payload)
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\json\__init__.py", line 301, in jsonify
    f"{dumps(data, indent=indent, separators=separators)}\n",
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\json\__init__.py", line 131, in dumps
    return _json.dumps(obj, **kwargs)
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\json\__init__.py", line 234, in dumps
    return cls(
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\json\encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\json\encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\json\__init__.py", line 50, in default
    return super().default(o)
  File "C:\Users\5927226\AppData\Local\Programs\Python\Python39\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type WindowsPath is not JSON serializable
ERROR:root:Object of type WindowsPath is not JSON serializable
```

File: asreviewcontrib/dataset_name/your_dataset.py [hunk 8]
```diff
@@ -1,20 +1,18 @@
...
+    """This is an example dataset group."""
 
-    __filepath__ = os.getcwd()
+    group_id = "example_group"
+    description = "Example dataset group"
 
-    def __init__(self):
-        """Initialize the dataset."""
-        super().__init__(
+    ExampleDatasetLocal = BaseDataSet(
```
Comment (J535D165) [inline asreviewcontrib/dataset_name/your_dataset.py]:
```suggestion
example_dataset_local = BaseDataSet(
```

File: setup.py [hunk 9]
```diff
@@ -17,20 +17,19 @@
...
+            "example_group = asreviewcontrib.dataset_name:ExampleDatasetGroup", # noqa
         ]
 
     },
-
     project_urls={
-        'Bug Reports': 'https://github.com/asreview/template-extension-new-dataset/issues',
-        'Source': 'https://github.com/asreview/template-extension-new-dataset/',
+        'Bug Reports': 'https://github.com/asreview/template-extension-new-dataset/issues', # noqa
+        'Source': 'https://github.com/asreview/template-extension-new-dataset/', # noqa
```
Comment (J535D165) [inline setup.py]: Is this required? If so, you need 2 spaces. 

```suggestion
        'Bug Reports': 'https://github.com/asreview/template-extension-new-dataset/issues',  # noqa
        'Source': 'https://github.com/asreview/template-extension-new-dataset/',  # noqa
```

**Review by J535D165 — APPROVED** [review]

**Pull request merged**